usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ParseSingleRowForFileTest.java

```diff
@@ -75,52 +92,132 @@ void shouldValidateAndParseOneByOne() {
+        when(mockFileRepo.findByName(anyString())).thenReturn(Optional.empty());
+        when(mockFileRepo.getProviderForFile(any(RawFileInfo.class)))
+                .thenAnswer(new Answer<Optional<RawFileRepository.RawEntityProvider>>() {
+                    public Optional<RawFileRepository.RawEntityProvider> answer(InvocationOnMock invocation) {
+                        RawFileInfo file = invocation.getArgument(0);
+                        if (file.getFilename().contains("empty")) {
+                            return Optional.empty();
+                        }
+
+                        if (file.getFilename().contains("invalid")) {
+                            return Optional.of(new EntityProvider(
+                                    List.of(
+                                            Map.of("header0_string", "header0_string",
+                                                    "header1_float", "header1_float",
+                                                    "header2_integer", "header2_integer"),
+                                            Map.of("header0_string", "invalid_string",
+                                                    "header1_float", "valid_float",
+                                                    "header2_integer", "invalid_integer"),
+                                            Map.of("header0_string", "valid", "header1_float", "invalid_float",
+                                                    "header2_integer", "valid_integer"),
+                                            Map.of("header0_string", "invalid", "header1_float", "invalid_float",
+                                                    "header2_integer", "invalid_integer")
+                                    )
+                            ));
+                        }
+
+                        return Optional.of(new EntityProvider(
+                                List.of(
+                                        Map.of("header0_string", "header0_string", "header1_float", "header1_float",
+                                                "header2_integer", "header2_integer"),
+                                        Map.of("header0_string", "valid_string", "header1_float", "valid_float",
+                                                "header2_integer", "valid_integer"),
+                                        Map.of("header0_string", "valid", "header1_float", "valid_float",
+                                                "header2_integer", "valid_integer"),
+                                        Map.of("header0_string", "valid", "header1_float", "invalid_float",
+                                                "header2_integer", "valid_integer")
+                                )
+                        ));
+                    }
+                });
+
+        return mockFileRepo;
+    }
+
+    private static class EntityParser implements GtfsSpecRepository.RawEntityParser {
```

Comment:
Could we use mockito `mock(` and then verify via `times(xx)` as done here for example
https://github.com/MobilityData/gtfs-validator/blob/74dae54474eefe81c007f001d95932bfe266e232/adapter/parser/src/test/java/org/mobilitydata/gtfsvalidator/parser/GtfsEntityParserTest.java#L80

Reply:
Verify and inOrder tests are now added.